From: Maor Gottlieb <maorg@mellanox.com>
Date: Fri, 10 Jun 2016 00:07:31 +0300
Subject: [PATCH] net/mlx5: Fix root flow table update
Patch-mainline: v4.7-rc3
Git-commit: 2fee37a47cebc26d58eec5dafc8ba787a6ee5350
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

When we destroy the last flow table we need to update
the root_ft to NULL.

It fixes an issue for when the last flow table is destroyed
and recreated again, root_ft pointer will not be updated,
as a result traffic will be dropped.

Fixes: 2cc43b494a6c ('net/mlx5_core: Managing root flow table')
Signed-off-by: Maor Gottlieb <maorg@mellanox.com>
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/fs_core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/fs_core.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/fs_core.c
@@ -1207,8 +1207,8 @@ static int update_root_ft_destroy(struct
 				       ft->id);
 			return err;
 		}
-		root->root_ft = new_root_ft;
 	}
+	root->root_ft = new_root_ft;
 	return 0;
 }
 
